modules/build/src/test/scala/scala/build/tests/BuildProjectTests.scala

```diff
@@ -0,0 +1,135 @@
+package scala.build.tests
+
+import scala.build.errors.Severity
+import com.eed3si9n.expecty.Expecty.expect
+import scala.build.options.{BuildOptions, InternalOptions, JavaOptions, Scope}
+import scala.build.{Build, Inputs, LocalRepo, Positioned, Sources}
+import scala.build.options.ScalaOptions
+import scala.build.Logger
+import scala.build.blooprifle.BloopRifleLogger
+import scala.build.errors.BuildException
+import scala.build.errors.Diagnostic
+import java.io.PrintStream
+import coursier.cache.CacheLogger
+import scala.build.Position
+
+class BuildProjectTests extends munit.FunSuite {
+
+  class LoggerMock extends Logger {
+
+    var diagnostics: List[Diagnostic] = Nil
+
+    override def message(message: => String): Unit = ???
+
+    override def log(s: => String): Unit = ???
+
+    override def log(s: => String, debug: => String): Unit = ???
+
+    override def debug(s: => String): Unit = {}
+
+    override def log(diagnostics: Seq[Diagnostic]): Unit = {
+      this.diagnostics = this.diagnostics ++ diagnostics
+    }
+
+    override def log(ex: BuildException): Unit = {}
+
+    override def exit(ex: BuildException): Nothing = ???
+
+    override def coursierLogger: CacheLogger = CacheLogger.nop
+
+    override def bloopRifleLogger: BloopRifleLogger = BloopRifleLogger.nop
+
+    override def scalaNativeLogger: scala.scalanative.build.Logger =
+      scala.scalanative.build.Logger.nullLogger
+
+    override def compilerOutputStream: PrintStream = ???
+
+  }
+
+  val bloopJavaPath = Position.Bloop("/home/empty/jvm/8/")
+
+  def testJvmReleaseIsSetCorrectly(
+    javaHome: String,
+    bloopJvmVersion: Int,
+    scalacOptions: Seq[String] = Nil
+  ) = {
+    val options = BuildOptions(
+      internal = InternalOptions(localRepository =
+        LocalRepo.localRepo(scala.build.Directories.default().localRepoDir)
+      ),
+      javaOptions = JavaOptions(
+        bloopJvmVersion =
+          Some(Positioned(bloopJavaPath, bloopJvmVersion)),
+        javaHomeOpt = Some(Positioned.none(os.Path(javaHome)))
+      ),
+      scalaOptions = ScalaOptions(scalacOptions = scalacOptions)
+    )
+
+    val inputs  = new Inputs(Nil, None, os.pwd, "project", false)
+    val sources = Sources(Nil, Nil, None, Nil, options)
+    val logger  = new LoggerMock()
+    val res     = Build.buildProject(inputs, sources, Nil, options, Scope.Test, logger)
+
+    val scalaCompilerOptions = res.right.get.scalaCompiler.scalacOptions
+    (scalaCompilerOptions, res.right.get.javacOptions, logger.diagnostics)
+  }
+
+  def jvm(v: Int) = os.proc("cs", "java-home", "--jvm", s"zulu:$v").call().out.text().trim()
```

Comment:
Ideally, we should use the [same value as in the ITs](https://github.com/alexarchambault/scala-cli/blob/a14bb70db2fc2c5190d512632e112d3133013712/modules/integration/src/test/scala/scala/cli/integration/TestUtil.scala#L63) (just `"cs"` is a problem when the launcher is `cs.bat` on Windows, but not when it's `cs.exe`, IIRC).